hardware_interface/src/component_parser.cpp

```diff
@@ -629,6 +654,218 @@ std::vector<HardwareInfo> parse_control_resources_from_urdf(const std::string &
     ros2_control_it = ros2_control_it->NextSiblingElement(kROS2ControlTag);
   }
 
+  // get min/max from ros2_control tag
+  auto retrieve_min_max_interface_values = [](const auto & itf, double & min, double & max) -> bool
+  {
+    try
+    {
+      if (itf.min.empty() && itf.max.empty())
+      {
+        // If the limits don't exist then return false as they are not retrieved
+        return false;
+      }
+      if (!itf.min.empty())
+      {
+        min = hardware_interface::stod(itf.min);
+      }
+      if (!itf.max.empty())
+      {
+        max = hardware_interface::stod(itf.max);
+      }
+      return true;
+    }
+    catch (const std::invalid_argument & err)
+    {
+      std::cerr << "Error parsing the limits for the interface: " << itf.name << " from the tags ["
+                << kMinTag << ": '" << itf.min << "' and " << kMaxTag << ": '" << itf.max
+                << "'] within " << kROS2ControlTag << " tag inside the URDF. Skipping it"
+                << std::endl;
+      return false;
+    }
+  };
+
+  // set custom values for acceleration and jerk limits (no URDF standard)
+  auto set_custom_interface_values =
+    [retrieve_min_max_interface_values](const auto & itr, joint_limits::JointLimits & limits)
+  {
+    if (itr.name == hardware_interface::HW_IF_ACCELERATION)
+    {
+      double max_decel(std::numeric_limits<double>::quiet_NaN()),
+        max_accel(std::numeric_limits<double>::quiet_NaN());
+      if (retrieve_min_max_interface_values(itr, max_decel, max_accel))
+      {
+        if (std::isfinite(max_decel))
+        {
+          limits.max_deceleration = std::fabs(max_decel);
+          if (!std::isfinite(max_accel))
+          {
+            limits.max_acceleration = std::fabs(limits.max_deceleration);
+          }
+          limits.has_deceleration_limits = true && itr.enable_limits;
```

Comment:
I noticed this pattern of `true &&` in a couple of places. Why?

Reply:
You are right, it can be simply `itr.enable_limits`. I will make the modifications soon and let you know

Reply:
You are right. This is fixed now. :+1: